features/image-transformations/overlay-using-layers.md

```diff
@@ -0,0 +1,200 @@
+
+* [Add images over image](overlay-using-layers#add-images-over-image)
+* [Add text over image](overlay-using-layers#add-text-over-image)
+* [Add solid color blocks over image](overlay-using-layers#add-images-over-image)
+
+# Layers
+A layer is a special kind of transformation that allows you to modify the overlay itself and express its position in relation to the parent.
+
+## Syntax of layers
+
+A layer starts with `l-<type>` and ends with `l-end`. All the positional and transformation parameters of that layer are between `l-<type>` and `l-end` and only apply to that layer and not the parent base asset.
+
+`type` can be `image` or `text`.
+
+For example, in the following URL, we are adding a logo image `logo.png` on top of a base image, i.e. `sample-image.jpg`. However, we applied width (`w-10`) and rotation (`rt-90`) transformations on this overlay logo image before placing it on the base image. Transformations `w-300,h-300` are applied to `sample-image.jpg`.
+
+Here the parent base image has one layer inside it. A layer can also nest another layer.
+
+```markup
+        URL-endpoint           Base image                                   Layer
+┌──────────────────────────┐┌──────────────┐                ┌─────────────────────────────────┐
+https://ik.imagekit.io/demo/sample-image.jpg?tr=w-300,h-300,l-image,i-logo.png,w-10,rt-90,l-end
+                                                └────┬────┘                    └────┬────┘
+                                                     │                              │
+                                                     │        These transformations are applied to logo.png
+                                                     │
+                           These transformations are applied to sample-image.jpg
+```
+
+## Input of layer
+The input of the layer can be specified using `i` or `ie` (base64 encoded) parameter. In case both `i` and `ie` is present, `i` is ignored. The base64 string should be made URL safe to ensure that all padding characters (=) are included correctly. In Javascript, a function like `encodeURIComponent()` can be used for this.
+
+## Position of layer
+
+The position of the layer can be controlled using the following parameters. The position of the layer is always relative to the immediate parent. For instance, the parent is the base image in the above example, and the logo image is the nested layer. 
+
+| Parameter   | Description |
+| ----------- | ----------- |
+| lx          | `x` of the top-left corner in the base asset where layer's top-left corner would be placed.    |
+| ly          | `y` of the top-left corner in the base asset where layer's top-left corner would be placed.    |
+| lfo         | Position of layer in relative terms e.g. `center`, `top`, `left`, `bottom`, `right`, `top_left`, `top_right`, `bottom_left` and `bottom_right`. Default value is `center`.     |
```

Comment:
we should mention what would happen if both lfo and lx/ly are given

Reply:
Done